### PR TITLE
Backend - Revisit all BD and queries for soft delete

### PR DESCRIPTION
**Pull request by AminesCodes**
Revisit all BD and queries for soft delete
## Description

All tables now have a deleted column, instead of delete cascade for references, the cascade now is functions that run for other tables when there is a reference
## Motivation and Context

Data is preserved , even if a user deletes something, it's not completely lost
## How Has This Been Tested?

Postman + Postico
## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: backend/db/seed.sql [hunk 1]
```diff
@@ -102,58 +111,62 @@ CREATE TABLE events (
...
+    ('Django', 'django'),
+    ('Flask', 'Flask'),
+    ('Java', 'java'),
+    ('Swift', 'swift'),
+    ('Objective-C', 'objectivec'),    -- 10
+    ('Python', 'python'),
+    ('R', 'r'),
+    ('Scala', 'scala'),
+    ('HTML', 'html'),
+    ('CSS', 'scc'),    -- 15
```
Comment (joseph-p-pasaoa) [inline backend/db/seed.sql:159]: typo, parsed should be 'css'

File: backend/queries/admin.js [hunk 2]
```diff
@@ -36,13 +36,33 @@ const updateAdmin = async (id, firstName, lastName) => {
...
+}
+
+const deleteAdminByEmail = async (email, promise) => {
+    const deleteQuery = `
+        UPDATE administration 
+            SET deleted = NOW()
+            WHERE a_email = $1
+            RETURNING *
+    `  
+    if (promise) {
```
Comment (joseph-p-pasaoa) [inline backend/queries/admin.js:55]: i couldnt find you calling this with promise in this pull req. what is this conditional for?

Comment (AminesCodes) [reply backend/queries/admin.js:55]: I wasn't sure to use it or not but all the functions where I added promise as an argument are for simulating the delete cascade on the database, it's meant to push all the async request to the promises array and wait for the array (trigger all the requests at the same time)

File: backend/queries/fellows.js [hunk 3]
```diff
@@ -85,19 +87,38 @@ const updateFellow = async (userObj) => {
...
+  // promises.push(eventFollowsQueries.delete...(fId, true));
+  // promises.push(mentorPairsQueries.delete...(fId, true));
+
+  const response = await Promise.all(promises);
+  return response[0];
+}
+
+const deleteFollowByEmail = async (email, promise) => {
+  const fellow = await db.one('SELECT * FROM fellows WHERE f_email = $1', email);
+  console.log(fellow)
```
Comment (joseph-p-pasaoa) [inline backend/queries/fellows.js:117]: console.log

File: backend/queries/volunteers.js [hunk 4]
```diff
@@ -142,7 +146,30 @@ const updateVolunteer = async (user) => {
...
+  // promises.push(mentorPairsQueries.delete...(id, true));
+  promises.push(volunteerSkillsQueries.deleteVolunteerSkillsByVolunteerId(id, true));
+  
+  const result = await Promise.all(promises);
+  return await result[0];
+}
+
+const deleteVolunteerByEmail = async (email, promise) => {
+  const volunteer = await db.one('SELECT * FROM volunteers WHERE v_email = $1', email);
+  console.log(volunteer)
```
Comment (joseph-p-pasaoa) [inline backend/queries/volunteers.js:207]: console.log

**Pull request merged**